nautobot/dcim/views.py

```diff
@@ -4168,6 +4171,157 @@ class PowerFeedUIViewSet(NautobotUIViewSet):
+                weight=400,
+                label="Electrical Characteristics",
+                fields=["supply", "voltage", "amperage", "phase", "max_utilization"],
+                value_transforms={
+                    "voltage": [lambda v: f"{v}V"],
+                    "amperage": [lambda a: f"{a}A"],
+                    "max_utilization": [lambda p: f"{p}%"],
+                },
+            ),
+            object_detail.KeyValueTablePanel(
+                section=SectionChoices.RIGHT_HALF,
+                weight=200,
+                label="Connection",
+                context_data_key="connection_data",
+            ),
+        )
+    )
+
+    def get_extra_context(self, request, instance):
+        context = super().get_extra_context(request, instance)
+        if not instance:
+            return context
+
+        # Safe foreign key access
+        power_panel = getattr(instance, "power_panel", None)
+        rack = getattr(instance, "rack", None)
+        status = getattr(instance, "status", None)
+
+        powerfeed_data = {
+            "Power Panel": power_panel or helpers.HTML_NONE,
+            "Rack": rack or helpers.HTML_NONE,
+            "Type": format_html(
+                '<span class="label label-{}">{}</span>',
+                escape(instance.get_type_class() or "default"),
+                escape(instance.get_type_display() or "Unknown"),
+            ),
+            "Status": helpers.hyperlinked_object_with_color(status) if status else helpers.HTML_NONE,
+        }
+
+        connected_device_data = self._get_connected_device_data(instance)
+        powerfeed_data.update(connected_device_data)
+
+        context["powerfeed_connected_data"] = powerfeed_data
+        context["connection_data"] = self._get_connection_data(request, instance)
+        return context
+
+    def _get_connected_device_data(self, instance):
+        connected_device = self._get_connected_device_html(instance)
+        utilization_label = self._get_utilization_display(instance)
+
+        return {
+            "Connected Device": connected_device,
+            "Utilization (Allocated)": utilization_label,
+        }
+
+    def _get_connected_device_html(self, instance):
+        if instance.connected_endpoint and instance.connected_endpoint.parent:
+            parent = helpers.hyperlinked_object(instance.connected_endpoint.parent)
+            return format_html("{} ({})", parent, instance.connected_endpoint)
+        else:
+            return helpers.HTML_NONE
+
+    def _get_utilization_display(self, instance):
+        endpoint = getattr(instance, "connected_endpoint", None)
+        if not endpoint or not hasattr(endpoint, "get_power_draw"):
+            return format_html('<span class="text-muted">N/A</span>')
+
+        utilization = endpoint.get_power_draw()
+        if not utilization or "allocated" not in utilization:
+            return format_html('<span class="text-muted">N/A</span>')
```

Comment:
These too should be letting the UI framework handle these cases and automatically use the correct `placeholder` HTML.